src/parser.ts

```diff
@@ -393,14 +393,24 @@ export class Parser {
     return modifiers;
   }
 
-  public getParameterInfo(callSignature: ts.Signature) {
+  public getParameterInfo(callSignature: ts.Signature): Method['params'] {
```

Comment:
I think it would make sense to create s specific "types" and use that instead of `Method['params']` and in `Method` definition, something like

```ts
export interface MethodParameterType {
        name: string;
        required: boolean;
}
export interface MethodParameter { 
    name: string;
    description?: string | null;
    type: MethodParameterType;
 }
```

Honestly, I am working with typescript almost every day but had no idea something like that `Method['params']` is in this context even possible :) 

More importantly it's a good opportunity to think about these types once again.  I think the `required` attribute has nothing to do with the type of the param but with the param itself. Because the param is optional not the type. It would be similar to the `PropItem` then.

Reply:
Good points. I'll do this!

Reply:
Done, please check to see if it meets your expectations

Reply:
OK, somehow that didn't come to my mind, that there is already styleguidist which expects something else, sorry for that.

If the parameters passed to the `ArgumentRenderer` are just the `Method` fields then it seems that  there has to be `type: 'OptionalType'` in the `MethodParameterType` . But I am not really sure if I read that code properly because there is also `returns` which I have no idea how that's related to argument. Please, check that component yourself [ArgumentRenderer](https://github.com/styleguidist/react-styleguidist/blob/34f3c83e769542157c72d0e055ad8850d22b6001/src/client/rsg-components/Argument/ArgumentRenderer.js)

Reply:
Yes, I tried the `OptionalType` approach, but that displays the type as optional, rather than the parameter, i.e. it would render `colIdx: number?` rather than `colIdx?: number`.

Reply:
Or maybe that's what we want? Because there's also the `defaultValue` field there which we should probably also populate.